Zotero/Scenes/Detail/Annotation Filter Popover/Views/AnnotationsFilterViewController.swift

```diff
@@ -11,165 +11,219 @@ import UIKit
+                container.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -Self.horizontalInset)
+            ])
+        }
 
-    // MARK: - Actions
+        func setupNavigationBar() {
+            let closeBarButtonItem = UIBarButtonItem(title: L10n.close, style: .plain, target: nil, action: nil)
+            closeBarButtonItem.rx.tap
+                .subscribe(onNext: { _ in
+                    close()
+                })
+                .disposed(by: disposeBag)
+            navigationItem.leftBarButtonItem = closeBarButtonItem
 
-    private func update(state: AnnotationsFilterState) {
-        if state.changes.contains(.colors) {
-            self.setSelected(colors: state.colors)
-        }
+            setupClearButton(visible: (!viewModel.state.colors.isEmpty || !viewModel.state.tags.isEmpty))
 
-        if state.changes.contains(.tags) {
-            self.set(tags: state.tags, availableTags: state.availableTags)
+            func close() {
+                updateFilter()
+                navigationController?.presentingViewController?.dismiss(animated: true)
+            }
         }
 
-        if state.changes.contains(.tags) || state.changes.contains(.colors) {
-            self.setupClearButton(visible: (!state.colors.isEmpty || !state.tags.isEmpty))
-        }
+        func update(state: AnnotationsFilterState) {
+            if state.changes.contains(.colors) {
+                setSelected(colors: state.colors)
+            }
 
-        self.updateFilter()
-        self.updatePreferredContentSize()
-    }
+            if state.changes.contains(.tags) {
+                set(tags: state.tags, availableTags: state.availableTags)
+            }
 
-    private func updatePreferredContentSize() {
-        let labelSize = self.tagsLabel.systemLayoutSizeFitting(CGSize(width: AnnotationsFilterViewController.width - 40, height: .greatestFiniteMagnitude))
-        let size = CGSize(width: AnnotationsFilterViewController.width, height: labelSize.height + 88) // 68 for circles, 20 bottom inset
-        self.preferredContentSize = size
-        self.navigationController?.preferredContentSize = size
-    }
+            if state.changes.contains(.tags) || state.changes.contains(.colors) {
+                setupClearButton(visible: (!state.colors.isEmpty || !state.tags.isEmpty))
+            }
 
-    private func setSelected(colors: Set<String>) {
-        for view in self.colorContainer.arrangedSubviews {
-            guard let pickerView = view as? ColorPickerCircleView else { continue }
-            pickerView.isSelected = colors.contains(pickerView.hexColor)
+            updateFilter()
+            updatePreferredContentSize()
         }
-    }
 
-    private func set(tags: Set<String>, availableTags: [Tag]) {
-        if availableTags.isEmpty {
-            self.tagsContainer.isHidden = true
-            return
+        func setSelected(colors: Set<String>) {
```

Comment:
At this point you can remove this function and just call `colorPicker?.setSelected(hexColors:)` directly.

Reply:
Check https://github.com/zotero/zotero-ios/pull/807/commits/2fae4dd91e37662605dfed6ee1237574ead4ea68